File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        int sLength = s.length();
        int left = 0;
        int right = 0;
        int maxSize = 0;
        Set<Character> sHash = new HashSet<>();

        while (right < sLength) {
            if (sHash.contains(s.charAt(right))){
                while(sHash.contains(s.charAt(right))) {
                    sHash.remove(s.charAt(left));
                    left++;
                }
            } else {
                sHash.add(s.charAt(right));
                right++;
            }

            maxSize = Math.max(right-left, maxSize);
        }

        return maxSize;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return 1
        
        dp = [1] * len(nums)
        dp[0] = 1

        for i in range(len(nums)):
            for j in range(i):
                if nums[j] < nums[i]:
                    dp[i] = max(dp[i], dp[j] + 1)

        return max(dp)